src/Avalonia.Base/Media/MediaInvalidation.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System.Collections.Generic;
+using Avalonia.Reactive;
+
+#nullable enable
+
+namespace Avalonia.Media
+{
+    /// <summary>
+    /// A system which consumes changes to <see cref="AvaloniaObject"/> instances which exist outside the visual tree, locates the <see cref="Visual"/> objects which 
+    /// depend on them for their rendered appearance, and calls the <see cref="Visual.InvalidateVisual"/> method on each one.
+    /// </summary>
+    public static class MediaInvalidation
+    {
+        private static readonly HashSet<AvaloniaProperty> s_registeredProperties = new();
+
+        /// <summary>
+        /// Marks one or more instances of <see cref="AvaloniaProperty"/> as affecting the appearance of the owner type.
+        /// </summary>
+        public static void AffectsMediaRender(params AvaloniaProperty[] properties)
+        {
+            foreach (var property in properties)
+            {
+                if (s_registeredProperties.Add(property))
+                {
+                    property.Changed.Subscribe(property.PropertyType.IsValueType ? OnMediaRenderPropertyChanged_Struct : OnMediaRenderPropertyChanged);
```

Comment:
nit: Subscribe here will create new instance of AnonymousObserver for property registration.
See https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Reactive/Observable.cs#L22
Since OnMediaRenderPropertyChanged_Struct and OnMediaRenderPropertyChanged are static, only two global AnonymousObservers can be preallocated.